chromium_src/chrome/browser/chrome_browser_main.cc

```diff
@@ -6,8 +6,143 @@
 #include "chrome/browser/chrome_browser_main.h"
 #include "brave/browser/brave_browser_process_impl.h"
 
+#include "base/command_line.h"
+#include "base/containers/fixed_flat_map.h"
+#include "base/strings/string_number_conversions.h"
+#include "base/strings/string_split.h"
+#include "build/build_config.h"
+#include "chrome/browser/profiles/profile.h"
+#include "chrome/browser/themes/theme_service.h"
+#include "chrome/browser/themes/theme_service_factory.h"
+#include "chrome/common/chrome_switches.h"
+#include "ui/base/mojom/themes.mojom.h"
+
+namespace {
+
+class ThemeCommandLineHandler {
+ public:
+  static void ProcessThemeCommandLineSwitches(
+      const base::CommandLine* command_line,
+      ThemeService* theme_service) {
+#if BUILDFLAG(IS_WIN) || BUILDFLAG(IS_MAC) || BUILDFLAG(IS_LINUX) || \
+    BUILDFLAG(IS_CHROMEOS)
+    if (!command_line || !theme_service) {
+      return;
+    }
+
+    if (command_line->HasSwitch(switches::kSetDefaultTheme)) {
+      theme_service->UseDefaultTheme();
+      return;
+    }
+
+    if (command_line->HasSwitch(switches::kSetUserColor)) {
```

Comment:
I see there are a few different switches being introduced, but they don't early return. Could end up with one switching override the behaviour of another switch if several of them are passed? Should we care about that?